inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/detection_output/detection_output_kernel_ref.cpp

```diff
@@ -0,0 +1,249 @@
+        MakeJitConstant("CONF_SIZE_Y", detectOutParams.conf_size_y),
+        MakeJitConstant("CONF_PADDING_X", detectOutParams.conf_padding_x),
+        MakeJitConstant("CONF_PADDING_Y", detectOutParams.conf_padding_y),
+        MakeJitConstant("SHARE_LOCATION", detectOutParams.share_location),
+        MakeJitConstant("VARIANCE_ENCODED_IN_TARGET", detectOutParams.variance_encoded_in_target),
+        MakeJitConstant("NMS_THRESHOLD", detectOutParams.nms_threshold),
+        MakeJitConstant("ETA", detectOutParams.eta),
+        MakeJitConstant("CONFIDENCE_THRESHOLD", detectOutParams.confidence_threshold),
+        MakeJitConstant("IMAGE_WIDTH", detectOutParams.input_width),
+        MakeJitConstant("IMAGE_HEIGH", detectOutParams.input_heigh),
+        MakeJitConstant("DECREASE_LABEL_ID", detectOutParams.decrease_label_id),
+        MakeJitConstant("CLIP_BEFORE_NMS", detectOutParams.clip_before_nms),
+        MakeJitConstant("CLIP_AFTER_NMS", detectOutParams.clip_after_nms),
+        MakeJitConstant("ELEMENTS_PER_THREAD", detectOutParams.elements_per_thread),
+        MakeJitConstant("PRIOR_COORD_OFFSET", detectOutParams.prior_coordinates_offset),
+        MakeJitConstant("PRIOR_INFO_SIZE", detectOutParams.prior_info_size),
+        MakeJitConstant("PRIOR_IS_NORMALIZED", detectOutParams.prior_is_normalized),
+    });
+
+    return jit;
+}
+
+int GetPartitionStep(int localWorkItemNum) {
+    int step_size = 0;
+    for (int temp = localWorkItemNum; temp > 1; temp /= 2) {
+        step_size++;
+    }
+    return step_size;
+}
+
+size_t GetOptimalLocalClassSize(std::vector<size_t> gws, const EngineInfo& info) {
+    const size_t optimal_values[] = {16, 8, 7, 6, 5, 4, 2, 1};
+    const size_t splitNum = gws[2];
+    const size_t globalClassNum = gws[1];
+    const auto rest_lws = info.maxWorkGroupSize / splitNum;
+    size_t lws_idx = 0;
+    while (rest_lws < optimal_values[lws_idx]) lws_idx++;
+    while (globalClassNum % optimal_values[lws_idx]) lws_idx++;
+
+    return optimal_values[lws_idx];
+}
+
+DetectionOutputKernelRef::DispatchData SetDefault(const detection_output_params& params, int idx) {
+    DetectionOutputKernelRef::DispatchData dispatchData;
+    const auto& input = params.inputs[0];
+    const auto& detectOutParams = params.detectOutParams;
+    constexpr size_t prior_box_size = 4;
+    auto loc_feature_num = params.inputs[0].Feature().v;
+    auto num_classes = detectOutParams.num_classes;
+    auto num_loc_classes = (detectOutParams.share_location) ? 1 : num_classes;
+    auto num_prior_boxes = (loc_feature_num / (num_loc_classes * prior_box_size));
+
+    if (idx == 0) {
+        if (detectOutParams.decrease_label_id) {
+            dispatchData.gws = {input.Batch().v, num_prior_boxes, 1};
+            dispatchData.lws = {1, 1, 1};
+        } else {
+            if (detectOutParams.conf_padding_x || detectOutParams.conf_padding_y) {
+                dispatchData.gws = {num_classes, 256, input.Batch().v};
```

Comment:
Why do you limit gws[1] by 256? I believe that processing of boxes can be done independently, if you just put some marker(`-1` or something like that) instead of elimination of box.

Reply:
Changed it to params.engineInfo.maxWorkGroupSize